src/main/services/torrent-client.ts

```diff
@@ -142,6 +143,83 @@ export class TorrentClient {
             }),
           }).show();
         }
+
+        if (userPreferences?.ShutDownAfterDownloadEnabled) {
+          const dialogPromise = dialog.showMessageBox({
+            type: "question",
+            buttons: ["Cancelar", "Confirmar"],
+            defaultId: 1,
+            title: "Confirmar desligamento",
+            message: "O sistema será desligado em breve. Tem certeza de que deseja prosseguir?",
+            detail: "Se você estiver usando o computador, clique em 'Cancelar' para abortar o desligamento, caso não selecione nada dentro de alguns segudos o desligamento ocorrerá",
+            cancelId: 0,
+            noLink: true,
+            normalizeAccessKeys: false,
+          });
+          
+          const timer = setTimeout(() => {
+            userPreferencesRepository.update(
+              { id: 1 },
+              { ShutDownAfterDownloadEnabled: false }
+            ).catch(error => {
+              console.error("Erro ao redefinir o valor de ShutDownAfterDownloadEnabled:", error);
+            });
+        
+            let shutdownCommand;
+            switch (os.platform()) {
```

Comment:
Using `process.platform` is more efficient in this case.
This switch-case block could be easily replaced by an object literal:
```ts
const command = {
'darwin': 'sudo shutdown -h +4'
}

command[process.platform]
```